target_chains/cosmwasm/tools/src/chains-manager/cosmwasm.ts

```diff
@@ -0,0 +1,240 @@
+import { DirectSecp256k1HdWallet, EncodeObject } from "@cosmjs/proto-signing";
+import {
+  ChainExecutor,
+  ExecuteContractRequest,
+  ExecuteContractResponse,
+  InstantiateContractRequest,
+  InstantiateContractResponse,
+  MigrateContractRequest,
+  MigrateContractResponse,
+  StoreCodeRequest,
+  StoreCodeResponse,
+  UpdateContractAdminRequest,
+  UpdateContractAdminResponse,
+} from "./chain-executor";
+import {
+  DeliverTxResponse,
+  MsgExecuteContractEncodeObject,
+  MsgInstantiateContractEncodeObject,
+  MsgStoreCodeEncodeObject,
+  SigningCosmWasmClient,
+  MsgMigrateContractEncodeObject,
+  MsgUpdateAdminEncodeObject,
+} from "@cosmjs/cosmwasm-stargate";
+import { GasPrice } from "@cosmjs/stargate";
+import Long from "long";
+import assert from "assert";
+
+export class CosmwasmExecutor implements ChainExecutor {
```

Comment:
is this the same "generic" version of the executor? if so, suggest naming both classes along the same lines so the reader understands the relationship.